docs/adrs/0080-use-flipt-feature-flags.md

```diff
@@ -0,0 +1,106 @@
+* Option #1: LaunchDarkly: This is the industry standard feature flag platform.
+* Option #2: Unleash: Open source, can be self-hosted
+* Option #3: AWS AppConfig: we're already using lots of AWS, what's a little more?
+* Option #4: Roll our own
+* Option #5: Flipt: Open source, self hosted, supports file based config
+
+## Decision Outcome
+
+Chosen Alternative: Option #5: Flipt
+
+LaunchDarkly is the industry standard, but there is no room for it in the budget. Unleash has similar features to AWS AppConfig, but requires more setup and maintenance effort when self-hosted (e.g. it requires a PostgreSQL database). AWS AppConfig is really a thin shim over config files and doesn't provide much in the way of helping us manage feature groupings.
+
+Rolling our own is a possibility, but one key feature we'd want to think about is how to promote feature flags from one environment to another (e.g. from staging to production). That suggests that tying the feature flag to the environment by e.g. storing the flags in the database isn't ideal as we need to recreate the flag settings in each environment. Combining that with the desire to have feature flags be enabled on a per user basis, it makes the complexity of rolling our own outweigh the relatively simple flipt deployment.
+
+Flipt provides a [filesystem backend](https://www.flipt.io/docs/experimental/filesystem-backends) which would allow a way for us to manage our feature flags using a [gitops](https://about.gitlab.com/topics/gitops/) style process. We can test out our flag configuration in separate environments (e.g. experimental, demo, staging) before rolling out to production. It also allows us to deploy the Flipt service without requiring another stateful system (e.g. no database).
+
+We will have a couple of different options for how we deploy flipt. Examining the options for how flipt is deployed may be its own ADR.
+
+## Pros and Cons of the Alternatives
+
+### Option #0: Do Nothing
+* '+' No change to App Eng workflow
+* '+' No additional setup for Platform Team
+* '-' We will at some point ship the wrong code
+* '-' Will break API versioning required by Prime
+
+
+### Option #1: LaunchDarkly
+* '+' Industry standard
+* '+' Feature Rich
+* '-' Expensive. No room in ODC budget
+* '-' Might not be allowed (It is FedRAMP moderate, not IL4)
+* '-' Unknown if Platform Team will need to be involved
+
+
+### Option #2: Unleash
+* '+' Open Source
+* '+' Can be self-hosted
+* '-' Adds another thing for Platform to do before Milestone 1
+* '-' Open source version may be limited
+* '-' Somewhat complicated deployment model
+
+
+### Option #3: AWS AppConfig
+* '+' Not a lot of additional setup, App Eng already used to getting other configuration information from AWS
+* '+' Very little Platform setup, AppEng already has limited access to AWS console
+* '+' Cost is rolled into existing AWS ODC, no separate approval
+* '-' Unknown cost
```

Comment:
**question:** How are both of these statements true?

Reply:
Thank you. I tried to reword to clarify